lighthouse.test.js

```diff
@@ -12,26 +31,37 @@ const output = scores => {
 }
 
 function launchChromeAndRunLighthouse(url, opts = {}, config = null) {
-  return chromeLauncher.launch({chromeFlags: opts.chromeFlags}).then(chrome => {
-    opts.port = chrome.port;
-    return lighthouse(url, opts, config).then(results => {
-      return chrome.kill().then(() => results)
-    });
-  });
+  return chromeLauncher
+    .launch({ chromeFlags: opts.chromeFlags })
+    .then(chrome => {
+      opts.port = chrome.port
+      return lighthouse(url, opts, config).then(results => {
+        return chrome.kill().then(() => results)
+      })
+    })
 }
 
 test('performance audit', async () => {
-  const { lhr } = await launchChromeAndRunLighthouse('http://localhost:9000')
-  
-  const scores = Object.keys(lhr.categories).reduce((merged, category) => {
-    merged[category] = lhr.categories[category].score
-    return merged
-  }, {})
-
-  console.log(output(scores))
-
-  expect(scores.performance).toBe(1)
-  expect(scores.accessibility).toBe(1)
-  expect(scores['best-practices']).toBeGreaterThanOrEqual(0.93)
-  expect(scores.seo).toBe(1)
-})
+  const serverIsAlive = await checkConnection('localhost', 9000).catch(err => {
+    console.warn(
+      'Performance not tested due to server not being available',
+      err
+    )
+  })
+  if (serverIsAlive) {
+    const { lhr } = await launchChromeAndRunLighthouse('http://localhost:9000')
+
+    const scores = Object.keys(lhr.categories).reduce((merged, category) => {
+      merged[category] = lhr.categories[category].score
+      return merged
+    }, {})
+
+    console.log(output(scores))
+
+    expect(scores.performance).toBe(1)
+    expect(scores.accessibility).toBe(1)
+    expect(scores['best-practices']).toBeGreaterThanOrEqual(0.93)
+    expect(scores.seo).toBe(1)
+  }
+  return
```

Comment:
Don’t need to return! Otherwise this looks great

Reply:
done 😉